clippy_lints/src/wildcard_let.rs

```diff
@@ -0,0 +1,39 @@
+use clippy_utils::diagnostics::span_lint_and_help;
+use rustc_ast::ast::{Local, PatKind};
+use rustc_lint::{EarlyContext, EarlyLintPass, LintContext};
+use rustc_middle::lint::in_external_macro;
+use rustc_session::impl_lint_pass;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// check for `let _ = ...`.
+    ///
+    /// this may be used by crates that with to force `#[must_use]`
+    /// values to actually used, along with `#[forbid(unused_must_use)]`.
```

Comment:
Is this use-case not already covered by [`let_underscore_must_use`](https://rust-lang.github.io/rust-clippy/master/index.html#/let_underscore_must_use)?

Reply:
in the 5 years of #4090 being open, noone brought that lint up.

Reply:
So I understand there is some overlap, but the wildcard_let lint is broader in scope, correct? In that case, the lint should at least check if the `let_underscore_must_use` lint would apply and avoid linting in that case to avoid double messages.